src/hasSubObject.test.ts

```diff
@@ -96,15 +96,35 @@ describe("typing", () => {
       const obj = {} as { a?: string; b?: number };
 
       if (hasSubObject(obj, {})) {
+        expectTypeOf(obj).toEqualTypeOf<
+          HasSubObjectBrand & { a?: string; b?: number }
+        >();
```

Comment:
i'm hesitant to include a brand type, because it'd impact how it works with the rest of the library. for example, in this branch, if we had `keys(obj)`, its type would be `Array<"a" | "b" | "__remeda">`, rather than the expected `Array<"a" | "b">`. i don't think the tradeoff here is worth it

Reply:
@cjquines Yeah, I thought about this but couldn't come up with a real world example where it would affect the user. Anyway, I agree that it might be too much adding type guard so feel free to close PR. I couldn't come up with a solution without a brand type.

Reply:
are your changes in #686 independent of this?

my vote is a weak no, but i'd also like to hear @eranhirsch 's thoughts

Reply:
> are your changes in https://github.com/remeda/remeda/pull/686 independent of this?

@cjquines Yes, they are. https://github.com/remeda/remeda/pull/686 handles a lot of cases but also might be too much for such function.